src/mo_frontend/typing.ml

```diff
@@ -982,6 +982,16 @@ and infer_exp'' env exp : T.typ =
       ot := t
     end;
     T.text
+  | ToCandidE exps ->
+    if not env.pre then begin
+        let ts = List.map (infer_exp env) exps in
+        if not (T.shared (T.seq ts)) then
+          error env exp.at "MXXXX" "to_candid argument must have shared type, but instead has non-shared type %a"
```

Comment:
You'll need to choose fresh error codes for the MXXXX placeholders, and either make sure some test flags them or add them to a list of ignored error codes by following instructions. Writing a test is easier...

Maybe just grep for an existing code to see how...

Reply:
Ah, I forgot that this was remaining.  Will do.

Reply:
Done.

I also recall a CI process that checks for failure test coverage.  Next I'll look at how to do negative failing tests to cover each new error number.

Reply:
test/fail is the natural home